libs/features/personalization/personalization.js

```diff
@@ -521,64 +693,80 @@ export async function runPersonalization(info, config) {
     blocks: selectedVariant.useblockcode,
-    fragments: selectedVariant.replacefragment,
+    fragments: selectedVariant.fragments,
   };
 }
 
-function cleanManifestList(manifests) {
-  const manifestPaths = [];
-  const cleanedList = [];
+function compareExecutionOrder(a, b) {
+  if (a.executionOrder === b.executionOrder) return 0;
+  return a.executionOrder > b.executionOrder ? 1 : -1;
+}
+
+export function cleanAndSortManifestList(manifests) {
+  const manifestObj = {};
   manifests.forEach((manifest) => {
-    try {
-      const url = new URL(manifest.manifestPath);
-      manifest.manifestPath = url.pathname;
-    } catch (e) {
-      // do nothing
-    }
-    const foundIndex = manifestPaths.indexOf(manifest.manifestPath);
-    if (foundIndex === -1) {
-      manifestPaths.push(manifest.manifestPath);
-      cleanedList.push(manifest);
+    if (!manifest) return;
+    manifest.manifestPath = normalizePath(manifest.manifestUrl || manifest.manifest);
+    if (manifest.manifestPath in manifestObj) {
+      let fullManifest = manifestObj[manifest.manifestPath];
+      let freshManifest = manifest;
+      if (manifest.name) {
+        fullManifest = manifest;
+        freshManifest = manifestObj[manifest.manifestPath];
+      }
+      freshManifest.name = fullManifest.name;
+      freshManifest.selectedVariantName = fullManifest.selectedVariantName;
+      freshManifest.selectedVariant = freshManifest.variants[freshManifest.selectedVariantName];
+      manifestObj[manifest.manifestPath] = freshManifest;
     } else {
-      cleanedList[foundIndex] = { ...cleanedList[foundIndex], ...manifest };
+      manifestObj[manifest.manifestPath] = manifest;
+      manifestObj[manifest.manifestPath] = manifest;
     }
   });
-  return cleanedList;
+  return Object.values(manifestObj).sort(compareExecutionOrder);
 }
 
-const createDefaultExperiment = (manifest) => ({
-  disabled: manifest.disabled,
-  event: manifest.event,
-  manifest: manifest.manifestPath,
-  variantNames: ['all'],
-  selectedVariantName: 'default',
-  selectedVariant: { commands: [] },
-});
+export function handleFragmentCommand(command, a) {
+  const config = getConfig();
+  const { action, fragment, manifestPath } = command;
+  if (action === 'replace') {
+    a.href = fragment;
+    if (config.mep.preview) a.dataset.manifestId = manifestPath;
+    return fragment;
+  }
+  if (action === 'remove') {
+    if (config.mep.preview) {
+      a.parentElement.dataset.removedManifestId = manifestPath;
+    } else {
+      a.parentElement.remove();
+    }
+  }
+  return false;
+}
 
 export async function applyPers(manifests) {
   const config = getConfig();
 
   if (!manifests?.length) return;
+  let experiments = manifests;
+  for (let i = 0; i < experiments.length; i += 1) {
+    experiments[i] = await getPersConfig(experiments[i], config.mep?.override);
+  }
 
-  const cleanedManifests = cleanManifestList(manifests);
+  experiments = cleanAndSortManifestList(experiments);
```

Comment:
**cleanManifestList** used to be triggered before the **getPersConfig** , was the order changed on purpose?
the reason I am asking is that now the MEP Preview can't disable the scheduled in future manifests:

(there should be no Fragment4)
[Before](https://main--milo--adobecom.hlx.page/drafts/mariia/pr/promo/promo?mep=%2Fdrafts%2Fmariia%2Fpr%2Fpromo%2Fpromomanifest.json--default%2C%2Fdrafts%2Fmariia%2Fpr%2Fpromo%2Fpromomanifestfuture.json--default)
[After](https://mepmoreactions--milo--adobecom.hlx.page/drafts/mariia/pr/promo/promo?mep=%2Fdrafts%2Fmariia%2Fpr%2Fpromo%2Fpromomanifest.json--default---%2Fdrafts%2Fmariia%2Fpr%2Fpromo%2Fpromomanifestfuture.json--default)

<img width="1433" alt="image" src="https://github.com/adobecom/milo/assets/6617234/93efb95e-8888-48c6-90f3-045f4dbb4f04">

I debugged why it happens, the path of disabled manifest never gets normalized.
If we can't change the order back, there is another way to fix it.
In `preloadManifests` you can move the normalizing line above 'if' statement:
```
manifest.manifestPath = normalizePath(manifest.manifestPath);
if (!manifest.manifestData && manifest.manifestPath && !manifest.disabled) {
      loadLink(
        manifest.manifestPath,
        { as: 'fetch', crossorigin: 'anonymous', rel: 'preload' },
      );
    }
```